cmake/TestLibXml2ConstErrorStructuredErrorCallback.cmake

```diff
@@ -0,0 +1,91 @@
+# Function: Test_LibXml2_Const_Error_Structured_Error_Callback
+#
+# Test if LibXml2 uses const Error * for structured error callback function.
+#
+# CONST_ERROR_STRUCTURED_ERROR_CALLBACK - true if const Error * is required
+# for structuredErrorCallback.
+#
+
+function(Test_LibXml2_Const_Error_Structured_Error_Callback)
+
+  set(_VAR_NAME "CONST_ERROR_STRUCTURED_ERROR_CALLBACK")
+  set(_HASH_VAR_NAME "HASH_${_VAR_NAME}")
+
+  # hash the CMAKE_FLAGS passed and check cache to know if we need to rerun
+  string(MD5 cmake_flags_hash "${CMAKE_SHARED_LINKER_FLAGS}")
```

Comment:
Why use lower case for `cmake_flags_hash`? Same below with `test_project_dir`. This is inconsistent with what we do in our CMake scripts.

Reply:
[1180414](https://github.com/cellml/libcellml/pull/1274/commits/1180414d58800111c40dd7ae6daff8acd10f736d)